src/odemis/driver/test/focustracker_test.py

```diff
@@ -23,88 +23,108 @@
 from __future__ import division
 
 import logging
-import numpy
-import os
-import unittest
-
-from canopen import SdoCommunicationError
-
 from odemis import model
 from odemis.driver.focustracker import FocusTrackerCO
 from odemis.model import NotSettableError
+import os
+import time
+import unittest
+
 
 logging.basicConfig(level=logging.DEBUG)
+logging.basicConfig(format="%(asctime)s  %(levelname)-7s %(module)s:%(lineno)d %(message)s")
 
 TEST_NOHW = (os.environ.get("TEST_NOHW", 0) != 0)  # Default to Hw testing
 
+KWARGS = {"name": "Focus Tracker",
+          "role": "focus-position",
+          "node_idx": 2,
+          "channel": 'can0',
+          # "inverted": ["z"]
+}
 if TEST_NOHW:
-    KWARGS = {"name": "Focus Tracker", "role": "focus-tracker", "node_idx": 0x10, "channel": 'fake'}
-else:
-    KWARGS = {"name": "Focus Tracker", "role": "focus-tracker", "node_idx": 0x10, "channel": 'can0'}
+    KWARGS["channel"] = "fake"
 
 
 class TestFocusTrackerCO(unittest.TestCase):
     """Test the focus tracker functionality with the CANopen network."""
 
-    def setUp(self):
-        self.focus_tracker = FocusTrackerCO(**KWARGS)
-        self.kwargs = KWARGS
+    @classmethod
+    def setUpClass(cls):
+        cls.focus_tracker = FocusTrackerCO(**KWARGS)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.focus_tracker.terminate()
 
     def test_connection(self):
-        """Test that connection to the network is successful when connected, and unsuccessful when disconnected."""
+        """Test that connection to the network is unsuccessful when disconnected."""
         if not TEST_NOHW:
-            # If the network scanner sees a node at index 0x10, the focus tracker is connected.
-            self.assertTrue(0x10 in self.focus_tracker.network.scanner.nodes)
-            # When the node_idx is incorrect an empty node is added to the network, this then results in an
-            # SDOCommunicationError when we try to send a message to an object in the object dictionary.
-            with self.assertRaises(SdoCommunicationError):
-                FocusTrackerCO(name="Focus Tracker", role="focus_tracker", channel=self.kwargs['channel'],
+            # If the node_idx is not found, maybe the device is disconnected => raise HwError
```

Comment:
Could you update this comment to state that 0x12 is not the node_idx of the actual device. I took me a while to realize what this test does.